### DISCUSSION TITLE
Is there a way to toggle all diffs in highlighted section (e.g. unstaged changes)?

### DISCUSSION THREAD

**Discussion by arnold-c** [NeogitOrg/neogit#1262] [Q&A] [answered]
It would be helpful to be able to toggle all diffs in a section. Currently, hitting `<tab>` toggles viewing which files are in the section. It would be helpful if there were a keybind e.g. `<c-tab>` that toggles all the diffs within the files in that section. I couldn't find anything in the docs or API, but wanted to ask here instead of opening an issue in case I just missed it.

Thanks,

**Comment by CKolkey** [comment, marked as answer]
Hey! Maybe its a failure of the docs then - you can use `1`, `2`, `3`, and `4` to set the fold level of the current section your cursor is in :)

**Reply by arnold-c** [reply]
Thanks @CKolkey for such a quick response. My brain defaulted to just looking for instances of toggle as that was the description for diffs, but the fold terminology makes sense when I think about it at all. That being said, they aren't traditional folds that you can interact with the default keybindings, so maybe it's worth adding this to the hints ([here](https://github.com/NeogitOrg/neogit/blob/0cae7abc30cb91d661f28257c331fcb5b5198e31/lua/neogit/status.lua#L181), I believe)? Given each depth has it's own `map_name` (based on my interpretation of [this](https://github.com/NeogitOrg/neogit/blob/0cae7abc30cb91d661f28257c331fcb5b5198e31/lua/neogit/config.lua#L370)), I'm not sure what the best way to handle this is, but if it's as simple as adding something like 

```lua
...
local_hints = {
    ...
    hint_label("Depth1", "toggle depth 1"),
    hint_label("Depth2", "toggle depth 2"),
    hint_label("Depth3", "toggle depth 3"),
    hint_label("Depth4", "toggle depth 4"),
    ...
}
...
```

I'm happy to submit a PR.

**Reply by CKolkey** [reply]
I appreciate the enthusiasm, but I should note that while you're right _on the master branch_ that these are not real folds, on the _nightly_ branch they now are! It'll be released with neovim 0.10. 

I'm not super keen to add extra help text, but if the documentation could be improved I'm all for that :)

**Reply by arnold-c** [reply]
That's great RE folds. Maybe something as simple as adding a common workflow video showcase to the README illustrating the core functionality to compliment the text that already exists?

**Reply by CKolkey** [reply]
You know, I've been thinking about making something for a while. I'm planning to do a whole feature release thing when nvim 0.10 is released, it would be a good idea to include it in that.
